Revise documentation

IgnatBeresnev opened the pull request:
[Link for preview](https://ignatberesnev.github.io/static/dokka/docs/kotlinlang/overview.html).

Alternatively, download and install [Stardust plugin](https://helpserver.labs.jb.gg/help/stardust/installing-stardust.html). Open any `.md` file with documentation, for instance `gradle.md`. On the right sidebar you'll see `Stardust Article Preview`, open it and you'll see rendered documentation.

Changes:
* Renamed `docs` module to `mkdocs` to avoid confusion. You can ignore changes in `mkdocs` module, it's all just moved files.
* Added a new `docs` Stardust module for new documentation.

Fixes #2202
Fixes #1929
Fixes #653
Fixes #614
Fixes #486
Fixes #2269
Fixes #2598

IgnatBeresnev commented on docs/vars.list:
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<vars>
+    <var name="dokkaVersion"
+         value="1.7.20"
+         type="string"/>
+    <var name="kotlinVersion"
+         value="1.7.20"
+         type="string"/>
+</vars>
```
Question to the website / tech writers team: I see that coroutines have no `vars.list`, but they use vars throughout documentation.

As I understand it, there's a common `vars.list` that is used throughout all of kotlinlang? Do we need to remove this one from our project, or can it stay?

IgnatBeresnev commented on docs/topics/runners/gradle.md:
```diff
@@ -0,0 +1,539 @@
+By default, you will find generated documentation under `build/dokka/{format}` directory of your project.
+Output location, among other things, can be [configured](#configuration) separately.
+
+### Multi-project builds
```
I debated myself on the use of `Multi-project` vs `Multimodule` as there can be confusion.

Decided to go with `Multi-project` since this is a topic about Gradle, and Gradle calls it [multi-project](https://docs.gradle.org/current/userguide/multi_project_builds.html#sec:creating_multi_project_builds) and all "modules" are really subprojects (even in buildscripts: `subprojects {}`)

Not sure if it's worth addressing naming in a separate note.

IgnatBeresnev commented on docs/dokka.tree:
```diff
@@ -0,0 +1,24 @@
+                 start-page="quickstart.md">
+
+    <toc-element id="overview.md"/>
+    <toc-element toc-title="Runners">
```
Maybe there's a better word than "Runners" that covers both build system plugins and the CLI alternative? Coulnd't come up with anything else

zarechenskiy commented on docs/topics/runners/gradle.md:
```diff
@@ -0,0 +1,539 @@
+{type="note"}
+
+
+### Legacy plugin application
```
I believe we can omit this part, even for Kotlin Gradle plugin we don't write this. Users, who cannot use plugins DSL, probably, already know how to use this other method

IgnatBeresnev replied:
I like this idea very much, sir. Will fix

IgnatBeresnev commented on docs/topics/runners/gradle.md:
````diff
@@ -0,0 +1,539 @@
+
+```kotlin
+plugins {
+    id("org.jetbrains.dokka") version "%dokkaVersion%"
````
It's actually a webhelp placeholder that is supposed to take the version from `vars.list`, but I'm not sure if it works in codeblocks (it probably should). If it doesn't, I'll definitely use the current version :+1:

IgnatBeresnev commented on docs/topics/runners/gradle.md:
```diff
@@ -0,0 +1,539 @@
+
+## Generating documentation
+
+Dokka's Gradle plugin comes with [HTML](html.md), [Markdown](markdown.md) and [Javadoc](javadoc.md) formats built in,
```
Sure, will reflect stability here and in gradle/maven/cli sections

vmishenev commented on docs/topics/plugins/versioning.md:
```diff
@@ -0,0 +1,315 @@
+
+Versioning plugin has a number of optional configuration properties:
+
+| **Task**                             | **Description**                                                                                                                                                                               |
```
Not `taks`

IgnatBeresnev replied:
Indeed, thanks for noticing, fixed

vmishenev commented on docs/topics/plugins/versioning.md:
```diff
@@ -0,0 +1,315 @@
+With versioning plugin applied and configured, no other steps are needed: documentation can be built the usual way.
+
+Versioning plugin will add a `version.json` file to the output folder. This file will be used by the plugin to match
+versions and generate version navigation. If your previously generated documentation does not have that file, you
```
Versioning plugin adds a template of the version navigation into HTML as well.

IgnatBeresnev replied:
Yeah, the idea was to use `version.json` as an indicator of whether documentation is versioned - not to say "add this file if it's not present"

Rephrased it a little, hopefully it's better now

vmishenev commented on docs/topics/plugins/versioning.md:
```diff
@@ -0,0 +1,315 @@
+
+## Applying the plugin
+
+You can apply Versioning plugin the same way as other [Dokka plugins](plugins_introduction.md#applying-dokka-plugins):
```
Can you put a notice that in multimodule projects versioning plugin should be applied in all submodules?

IgnatBeresnev replied:
Indeed, fixed

IgnatBeresnev commented on docs/topics/runners/maven.md:
```diff
@@ -0,0 +1,227 @@
+To generate documentation for Maven-based projects, you can use [Dokka Maven plugin](#applying-the-plugin).
+
+> Compared to the [Gradle plugin](gradle.md), Maven plugin has only basic features and 
+> **does not support multiplatform projects**.
```
I copied it from the old documentation, thinking it was still possible 

Removed the note about multiplatform support :+1:

IgnatBeresnev commented on docs/topics/formats/html.md:
```diff
@@ -0,0 +1,278 @@
+| `footerMessage`                         | Text displayed in the footer.                                                                                                                                                                           |
+| `separateInheritedMembers`              | Boolean property. If set to `true`, Dokka will render properties/functions and inherited properties/inherited functions separately. Disabled by default.                                                |
+| `templatesDir`                          | Path to the directory with custom HTML templates. See [templates](#templates) section for more details.                                                                                                 |
+| `mergeImplicitExpectActualDeclarations` | Boolean property. If set to true, Dokka will merge declarations that are not declared as expect/actual, but have the same fully qualified name. Can be useful in legacy codebases. Disabled by default. |
```
Added a link to the expect/actual topic on kotlinlang: https://kotlinlang.org/docs/multiplatform-connect-to-apis.html

IgnatBeresnev commented on docs/topics/runners/gradle.md:
```diff
@@ -0,0 +1,539 @@
+Output location, among other things, can be [configured](#configuration) separately.
+
+### Multi-project builds
+
```
Fixed

IgnatBeresnev commented on docs/topics/runners/gradle.md:
```diff
@@ -0,0 +1,539 @@
+Each subproject will have _partial_ tasks created for it: `dokkaHtmlPartial`,`dokkaGfmPartial`,
+and `dokkaJekyllPartial`. 
+
+These tasks are not intended to be used independently and exist only to be called by the parent's MultiModule task.
```
There's indeed many ways to configure tasks in Gradle. I don't think we should list all possible options, but we should give one simple and very common example and stick to it across all documentation

I don't think any particular way is better, the choice here was based on what I use and what I've seen used in other projects. We can discuss it if you think some other way should be used instead, I don't mind

IgnatBeresnev commented:
Addressed review comments in a separate commit, and added a `quickstart` section in overview

IgnatBeresnev commented:
I've updated the [preview](https://ignatberesnev.github.io/static/dokka/docs/kotlinlang/overview.html) to include descriptions of configuration options for Gradle, Maven and CLI runners, and made a final round of self-review.

This is the final version from me, now it's review fixes only :)

IgnatBeresnev commented:
* Fixed review comments that we got from the technical writers
* Updated the [preview](https://ignatberesnev.github.io/static/dokka/docs/kotlinlang/overview.html)
* Rebased onto master and resolved merge conflicts, but **had to force push**

sarahhaggarty reviewed (approved):
Fantastic work @IgnatBeresnev!
